src/v/pandaproxy/kafka_client_cache.h

```diff
@@ -60,6 +61,14 @@ class kafka_client_cache {
     size_t size() const;
     size_t max_size() const;
 
+    template<std::invocable<kafka_client_cache&> Func>
+    auto invoke_on_cache(Func&& func) {
+        return ss::try_with_gate(_gc_gate, [this, &func] {
```

Comment:
I think here, `func` should be moved or copied. So if you want to keep the (universal) rval as the parameter type, then you'd:
```
_gc_gate, [this, func = std::forward<Func>(func)] ....
```
Because its not guaranteed that a reference to it will still be alive between scheduling points

Reply:
I'll try this out but the only place that uses `invoke_on_cache` is the auth_ctx_server (See changes in server.h in this commit):
```cpp
            // Access the cache on the appropriate shard.
            return service().client_cache().invoke_on(
              user_shard(user.name),
              ss::smp_submit_to_options{context().smp_sg},
              [func{std::forward<Func>(func)}](
                kafka_client_cache& cache) mutable {
                  return cache.invoke_on_cache(std::forward<Func>(func));
              });
```